crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -588,33 +367,92 @@ impl PipelineCache {
                 }),
         };
 
-        let pipeline = self.device.create_render_pipeline(&descriptor);
+        let pipeline = device.create_render_pipeline(id, &descriptor);
 
-        CachedPipelineState::Ok(Pipeline::RenderPipeline(pipeline))
+        PipelineState::Ok(pipeline)
     }
 
-    fn process_compute_pipeline(
+    // TODO: elevate this method to remove the duplication
+    fn process_queue(
         &mut self,
-        id: CachedPipelineId,
+        device: &RenderDevice,
+        shader_cache: &mut ShaderCache,
+        layout_cache: &mut LayoutCache,
+    ) {
```

Comment:
This implementation is duplicated for both the `RenderPipeline` and the `ComptuePipeline` impl of `PipelineCacheInternal`. Is there a way to tell the type that the `process_pipeline` method exists for both?

Reply:
Fixed this